File: app/src/main/java/com/labessence/biotech/finaldilution/component/concentration/ConcentrationFactory.java
package com.labessence.biotech.finaldilution.component.concentration;

public class ConcentrationFactory {
    public static Concentration createConcentration(ConcentrationType type, double amount) {
        switch (type) {
            case MOLAR:
                return new MolarConcentration(amount);
            case MILIMOLAR:
                return new MilimolarConcentration(amount);
            case PERCENTAGE:
                return new PercentageConcentration(amount);
            case MILIGRAM_PER_MILLILITER:
                return new MgMlConcentration(amount);
        }
        return null;
    }
}

File: app/src/main/java/com/labwhisper/biotech/finaldilution/component/concentration/ConcentrationFactory.kt
package com.labwhisper.biotech.finaldilution.component.concentration

object ConcentrationFactory {
    fun createConcentration(type: ConcentrationType, amount: Double): Concentration {
        return when (type) {
            ConcentrationType.MOLAR -> MolarConcentration(amount)
            ConcentrationType.MILIMOLAR -> MilimolarConcentration(amount)
            ConcentrationType.PERCENTAGE -> PercentageConcentration(amount)
            ConcentrationType.MILIGRAM_PER_MILLILITER -> MgMlConcentration(amount)
            ConcentrationType.NX -> NXConcentration(amount)
        }
    }
}
